### DISCUSSION TITLE
Using visual selections in snippets?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [L3MON4D3/LuaSnip#1254] [Q&A]
For instance, I want to visually select text, and use a hotkey to trigger a snippet which surrounds the section with a markdown code block with my cursor at the end of the first line to write the language name. I looked through the documentation and all I found was `$TM_SELECTED_TEXT` but I couldn't get it working.

```lua
local mdCodeBlock =  s({trig = "````", name = "Codeblock"}, fmt([[
    ````{}
		$TM_SELECTED_TEXT
    ````
    ]], {
      i(1)
  }))
```

**Comment by L3MON4D3** [comment]
Oh, add something like `ls.setup({store_selection_keys="<Tab>"})` to your config (or, if you already have `ls.setup`, add the `store_selection_keys`-part). Then you can use `<Tab>` (or any other hotkey), and the next snippet should have `TM_SELECTED_TEXT` :)

**Reply by whoop-t** [reply]
> Oh, add something like `ls.setup({store_selection_keys="<Tab>"})` to your config (or, if you already have `ls.setup`, add the `store_selection_keys`-part). Then you can use `<Tab>` (or any other hotkey), and the next snippet should have `TM_SELECTED_TEXT` :)

Wont this delete the highlighted selection?

Is there a way to do it without deleting?

**Reply by L3MON4D3** [reply]
It will, but the selection can be reinserted immediately.
Why is that an issue? extmarks?

**Reply by whoop-t** [reply]
@L3MON4D3 

I'm trying to create a snippet where you can highlight something like a variable, then trigger the snippet and insert that variable into the snippet without deleting what is selected.

Like say I want to highlight a variable and then insert a print/log statement for that variable for debugging. Since the above way deletes/reinserts the word it doesn't really work without altering the line I'm trying to use for the snippet.

**Reply by L3MON4D3** [reply]
Ohhh okay, gotcha :)

You can use "$CLIPBOARD", it is a variable that contains text in the unnamed register, ie. the last yank.
Use it like `log("$CLIPBOARD", $CLIPBOARD)`
